### PR TITLE
Hide sidebar button

### PR DESCRIPTION
**Pull request by l2-**
can now collapse sidebar with the arrow button in the titlebar. also buttons in the titlebar are now added to left side of the existing buttons

### CODE REVIEW

**Comment by Kamielvf** [comment]
Any previews available? Also, how does it behave when the custom window chrome is disabled?

**Comment by l2-** [comment]
https://i.imgur.com/5Z6AxVb.gifv here you go

File: runelite-client/src/main/java/net/runelite/client/ui/TitleToolbar.java [hunk 1]
```diff
@@ -24,6 +24,15 @@
  */
 package net.runelite.client.ui;
 
+import lombok.extern.slf4j.Slf4j;
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/ui/TitleToolbar.java]: Don't reorder imports

File: runelite-client/src/main/java/net/runelite/client/ui/TitleToolbar.java [hunk 2]
```diff
@@ -179,4 +239,25 @@ public void remove(Component c)
...
+		tx.translate(-image.getWidth(null), 0);
+		AffineTransformOp op = new AffineTransformOp(tx, AffineTransformOp.TYPE_NEAREST_NEIGHBOR);
+		image = op.filter(image, null);
+		return image;
+	}
+
+	void flipArrowIcon()
+	{
+		SwingUtilities.invokeLater(() ->
+			{
```
Comment (devLotto) [inline runelite-client/src/main/java/net/runelite/client/ui/TitleToolbar.java]: Should use this format:

```java
SwingUtilities.invokeLater(() ->
{
    clientUi.getTitleToolbar().remove(titleBarButton);
});````

**Comment by deathbeam** [comment]
Maybe removing the toggling via navigation buttons here would be nice? Not sure. Also how would someone hide the sidebar with window chrome disabled as @Kamielvf said.

File: runelite-client/src/main/java/net/runelite/client/ui/TitleToolbar.java [hunk 3]
```diff
@@ -179,4 +238,25 @@ public void remove(Component c)
...
+	void flipArrowIcon()
+	{
+		SwingUtilities.invokeLater(() ->
+		{
+			sideBarIcon = horizontalFlipImage(sideBarIcon);
+			invertedIcon = horizontalFlipImage(invertedIcon);
+			sideBarButton.setIcon(new ImageIcon(sideBarIcon.getScaledInstance(ICON_SIZE, ICON_SIZE, Image.SCALE_SMOOTH)));
+			sideBarButton.setRolloverIcon(new ImageIcon(invertedIcon.getScaledInstance(ICON_SIZE, ICON_SIZE, Image.SCALE_SMOOTH)));
+		}
+		);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/TitleToolbar.java]: This should be on same line as  }

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 4]
```diff
@@ -82,10 +83,21 @@
...
 	private JPanel navContainer;
 	private PluginToolbar pluginToolbar;
+	private JPanel container;
+
+	@Getter
 	private PluginPanel pluginPanel;
 
+	@Getter
+	@Setter
+	private PluginPanel lastPluginPanel;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: Can this be done in better way without creating web of dependencies between everything like this?

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 5]
```diff
@@ -82,10 +83,21 @@
...
+
 	@Getter
 	private TitleToolbar titleToolbar;
+	
+	@Getter
+	@Setter
+	private boolean showSideBar = true;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: See my previous comment

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 6]
```diff
@@ -359,7 +371,7 @@ public void windowClosing(WindowEvent e)
 		pluginToolbar = new PluginToolbar(this);
 		container.add(pluginToolbar);
 
-		titleToolbar = new TitleToolbar(properties);
+		titleToolbar = new TitleToolbar(this, properties);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: See my previous comment

File: runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java [hunk 7]
```diff
@@ -410,6 +423,30 @@ void contract()
 		pluginPanel = null;
 	}
 
+	void expandSideBar()
+	{
+		if (showSideBar)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java]: You can't call this 2 times in a row right? so why is the boolean check needed here.

**Comment by l2-** [comment]
native chrome don't have the option to collapse sidebar buttons. as adam said if you put the button in a bar underneath the titlebar you're just moving the problem not solving it

**Comment by deathbeam** [comment]
Copy-pasta OSBuddy, problem solved! :D

File: runelite-client/src/main/java/net/runelite/client/ui/PluginToolbar.java [hunk 8]
```diff
@@ -104,6 +104,10 @@ private void onClick(NavigationButton button)
 			current.setSelected(true);
 
 			PluginPanel pluginPanel = panelSupplier.get();
+			if (!ui.isSideBarShown())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/ui/PluginToolbar.java]: You can check for this boolean directly in the expandSideBar method and just call the method directly

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/ui/PluginToolbar.java]: Maybe adding some boolean also to method param would do it

**Comment by deathbeam** [comment]
Support for hiding the sidebar was already added

**Pull request closed without merging**